helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.testing;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.net.HostAndPort;
+
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerCertificateException;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.DockerException;
+import com.spotify.docker.client.DockerRequestException;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ContainerCreation;
+import com.spotify.docker.client.messages.ContainerExit;
+import com.spotify.docker.client.messages.HostConfig;
+import com.spotify.docker.client.messages.Info;
+import com.spotify.docker.client.messages.PortBinding;
+import com.spotify.helios.client.HeliosClient;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Collections.singletonList;
+
+public class HeliosSoloDeployment implements HeliosDeployment {
+
+  private static final Logger log = LoggerFactory.getLogger(HeliosSoloDeployment.class);
+
+  public static final String BOOT2DOCKER_SIGNATURE = "Boot2Docker";
+  public static final String PROBE_IMAGE = "onescience/alpine:latest";
+  public static final String HELIOS_IMAGE = "spotify/helios-solo:latest";
+  public static final String HELIOS_NAME_PREFIX = "solo.local.";
+  public static final String HELIOS_CONTAINER_PREFIX = "helios-solo-container-";
+  public static final int HELIOS_MASTER_PORT = 5801;
+
+  final DockerClient dockerClient;
+  final DockerHost dockerHost;
+  final String namespace;
+  final List<String> env;
+  final List<String> binds;
+  final String heliosContainerId;
+  final HeliosClient heliosClient;
+
+  HeliosSoloDeployment(final Builder builder) {
+    final String heliosHost;
```

Comment:
another nit but prefer to declare vars as close to their point of use as possible, so in this case next to the assignment